docs/source/60_building_apps.md

```diff
@@ -284,3 +284,327 @@ class MyCommandKryoRegistrator extends KryoRegistrator {
+  * Autodetected from either of the above using `loadSequences` (Scala and Java)
+* Coverage data as a `CoverageRDD`:
+  * From Parquet using `loadParquetCoverage` (Scala only)
+  * From Parquet or any of the feature file formats using `loadCoverage` (Scala
+    only)
+* Contig sequence as a broadcastable `ReferenceFile` using `loadReferenceFile`,
+  which supports 2bit files, FASTA, and Parquet (Scala only)
+
+The `JavaADAMContext` class provides Java-friendly methods that are equivalent
+to the `ADAMContext` methods. Specifically, these methods use Java types, and do
+not make use of default parameters. In addition to the load/save methods
+described above, the `ADAMContext` adds the implicit methods needed for using
+[ADAM's Pipe API](#pipes).
+
+### Working with genomic data using GenomicRDDs {#genomic-rdd}
+
+As described in the section on using the [ADAMContext}(#adam-context), ADAM
+loads genomic data into a `GenomicRDD` which is specialized for each datatype.
+This `GenomicRDD` wraps Apache Spark's Resilient Distributed Dataset (RDD,
+[@zaharia12]) API with genomic metadata. The `RDD` abstraction presents an
+array of data which is distributed across a cluster. `RDD`s are backed by
+a computational lineage, which allows them to be recomputed if a node fails and
+the results of a computation are lost. `RDD`s are processed by running
+functional [transformations]{#transforming} across the whole dataset.
+
+Around an `RDD`, ADAM adds metadata which describes the genome, samples, or
+pipeline that a dataset came from. Specifically, ADAM supports the following
+metadata:
+
+* `GenomicRDD` base: A sequence dictionary, which describes the reference
+  assembly that data is aligned to, if it is aligned. Applies to all types.
+* `MultisampleGenomicRDD`: Adds metadata about the samples in a dataset.
+  Applies to `GenotypeRDD`.
+* `ReadGroupGenomicRDD`: Adds metadata about the read groups attached to a
+  dataset. Applies to `AlignmentRecordRDD` and `FragmentRDD`.
+
+Additionally, `GenotypeRDD`, `VariantRDD`, and `VariantContextRDD` store the
+VCF header lines attached to the original file, to enable a round trip between
+Parquet and VCF.
+
+`GenomicRDD`s can be transformed several ways. These include:
+
+* The [core preprocessing](#algorithms) algorithms in ADAM:
+  * Reads:
+    * Reads to coverage
+    * [Recalibrate base qualities](#bqsr)
+    * [INDEL realignment](#realignment)
+    * [Mark duplicate reads](#duplicate-marking)
+  * Fragments:
+    * [Mark duplicate fragments](#duplicate-marking)
+* [RDD transformations](#transforming)
+* [By using ADAM to pipe out to another tool](#pipes)
+
+#### Transforming GenomicRDDs {#transforming}
+
+Although `GenomicRDD`s do not extend Apache Spark's `RDD` class, `RDD`
+operations can be performed on them using the `transform` method. Currently,
+we only support `RDD` to `RDD` transformations that keep the same type as
+the base type of the `GenomicRDD`. To apply an `RDD` transform, use the
```

Comment:
Nit: Add an example: i.e. We can transform AlignmentRecordRDDs to increment the position of the AlignmentRecord by 1 for every AlignmentRecord, but we cannot transform an AlignmentRecordRDD to a FeatureRDD.

I think it will make it much more clear.